bmq-sdk/src/main/java/com/bloomberg/bmq/impl/infr/util/expressionvalidator/ExpressionValidator.java

```diff
@@ -49,125 +47,95 @@ public boolean validate(java.io.Reader expression) throws IOException {
         do {
             // Get token
             Token token = scanner.yylex();
-            if (!isValid(token)) return false;
 
-            switch (token.getType()) {
-                case LPAR:
-                    paranthesisCounter++;
-                    break;
-                case RPAR:
-                    if (paranthesisCounter == 0
-                            || (prevToken != null && prevToken.getType() == Token.Type.LPAR)) {
-                        errorMessage =
-                                "syntax error, unexpected \")\" at offset " + token.getPosition();
-                        return false;
+            Token.Type tokenType = token.getType();
+            if (tokenType == Token.Type.INVALID) { // Check syntax error detected by scanned
+                StringBuilder sb =
+                        new StringBuilder("syntax error, unexpected invalid character \"");
+                errorMessage =
+                        sb.append(token.getValue())
+                                .append("\" at offset ")
+                                .append(token.getPosition())
+                                .toString();
+                return new ValidationResult(false, errorMessage);
```

Comment:
Idea: make static constructing functions for `ValidationResult` to simplify its usage here, something like:

```
ValidationResult.makeSuccess();
ValidationResult.makeUnexpectedLiteral(token);
ValidationResult.makeIntegerOverflow(token);
ValidationResult.makeTooManyProperties();
ValidationResult.makeNoProperties();
ValidationResult.makeUnmatchedParanthesis();
```

We can hide string construction build in these methods. Also, if we don't want to expose token class to `ValidationResult`, we can pass just literal and its position.

Reply:
Added static methods to construct `ValidationResult`